tools/WrapExecutableLinker.sh

```diff
@@ -4,7 +4,7 @@
 # See LICENSE.txt for details.
 
 temp_files=()
-trap 'rm -r "${temp_files[@]}"' EXIT
+trap '[ -z $temp_files ] || rm -r "${temp_files[@]}"' EXIT
```

Comment:
I'm not completely sure what you're trying to do here, but this is almost certainly not it.  You probably just want to add `-f` to the `rm` options.  Otherwise, be explicit and write the test as `[ -z "${temp_files[0]}" ]` and add a comment explaining why it is there.

Reply:
`-f` seems to work 👍

Reply:
(Thinking about this, the previous state was actually a bug.  Your new stuff just triggered it reliably.)